server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Scanner.java

```diff
@@ -61,39 +61,27 @@ public ScanBatch read() throws IOException, TabletClosedException {
     Batch results = null;
 
     try {
+      scannerSemaphore.acquire();
+    } catch (InterruptedException e) {
+      sawException = true;
+    }
 
-      try {
-        scannerSemaphore.acquire();
-      } catch (InterruptedException e) {
-        sawException = true;
-      }
-
-      // sawException may have occurred within close, so we cannot assume that an interrupted exception was its cause
-      if (sawException)
-        throw new IllegalStateException("Tried to use scanner after exception occurred.");
+    // sawException may have occurred within close, so we cannot assume that an interrupted exception was its cause
+    if (sawException)
+      throw new IllegalStateException("Tried to use scanner after exception occurred.");
 
-      if (scanClosed)
-        throw new IllegalStateException("Tried to use scanner after it was closed.");
+    if (scanClosed)
+      throw new IllegalStateException("Tried to use scanner after it was closed.");
 
-      if (options.isIsolated()) {
-        if (isolatedDataSource == null)
-          isolatedDataSource = new ScanDataSource(tablet, options);
-        dataSource = isolatedDataSource;
-      } else {
-        dataSource = new ScanDataSource(tablet, options);
-      }
+    if (options.isIsolated()) {
+      if (isolatedDataSource == null)
+        isolatedDataSource = new ScanDataSource(tablet, options);
+      dataSource = isolatedDataSource;
+    } else {
+      dataSource = new ScanDataSource(tablet, options);
+    }
 
-      SortedKeyValueIterator<Key,Value> iter;
-
-      if (options.isIsolated()) {
-        if (isolatedIter == null)
-          isolatedIter = new SourceSwitchingIterator(dataSource, true);
-        else
-          isolatedDataSource.reattachFileManager();
-        iter = isolatedIter;
-      } else {
-        iter = new SourceSwitchingIterator(dataSource, false);
-      }
+    try (SortedKeyValueIterator<Key,Value> iter = createIter(dataSource)) {
```

Comment:
Tablet servers reuse open rfiles.  I am wondering if this change or the change to ScanDataSource will cause those reused RFiles to be closed.